### ISSUE TITLE
Significant delays in tranmission immediately after connect.

### ISSUE THREAD

**Issue by jarekwg** [Quobject/SocketIoClientDotNet#29]
Hiya,

This one's a two-parter. I'm not sure whether this issue belongs on your side, or socket.io@1.3.2's, but there seems to be a brief moment right after initial connection, where transmission is unreliable. The following are two cases that worked with SocketIO4Net&socket.io@0.9.17, but seem to have trouble now:

(a) Attempting to transmit any moderately-sized packet(s) from the server to the client on the server's "connect" event results in significant lag time (i've seen up to half a minute), before the "connect" event fires on the client side, after which the sent packet is immediately received. 
(b) The issue in (a) seems to be remedied for (the most part) if instead of sending anything on the server's "connect" event, we instead wait for the client's "connect" event, at which point we emit a request to the server, asking for the said moderately-sized packet(s) to be transmitted. However, on the INITIAL connect event (ie. when IO.Socket() is called for the first time in an application [.Close()ing it and calling a fresh IO.Socket() within the same application seems to not suffer from this]), anything emitted on the client's "connect" event seems to sit in the local buffer for a cool 25 seconds (or until it's nudged by some other client emit), before it's sent to the server. Is there anything special about the first time IO.Socket() is called that may be causing this? - like, setting up statics or something (sorry, I haven't looked at the source code as of yet).

cheers,
jarekwg

**Comment by jarekwg** [comment]
Found the issue for (b) !! :D

I'm using an emit wrapper:
public void emit(string eventName, dynamic payload) {
    m_socket.Emit(eventName, JObject.FromObject(payload);
}

Hence in my connection event i had:
emit("requestModel", new {});
It appears that there's something funky going on with the serialisation, since when i changed the requestModel call to avoid using JObject.FromObject() the problem disappeared. The problem also disappeared when i threw the emit call in a separate thread and made it wait at least 100ms before calling the emit.


To reiterate:
m_socket.Emit('example', new JObject()); //Works in all cases.
m_socket.Emit('example', JObject.FromObject(new {})) //Works in all cases except for the first ~100ms after the connect event fires.

So this appears to either be a Linq problem, or some weird interaction between SocketIOClientDotNet and Linq. I'm going to stop my exploration of this issue here, as I've got things running smoothly now in my application and lack the time to explore more deeply. This might be worth looking into for you though, since you're using the Linq library as a dependency. Else, just go ahead and close the issue.

Cheers,
Jarek
